docs/docs/remark-ast.md

```diff
@@ -0,0 +1,201 @@
+---
+title: Manipulating Remark ASTs
+---
+
+gatsby-transformer-remark empowers developers to translate Markdown into HTML to be consumed throughout Gatsby pages. Blogs and other content based sites can highly benefit from this so the code and content is separated. With this, people who will write content for the site don't need to worry about how the the site was written, but rather can focus just on writing the Markdown.
+
+In certain instances, a developer may want to customize certain Markdown nodes while being parsed. Examples could include [add syntax highlighting](/packages/gatsby-remark-prismjs/), [parse images](/packages/gatsby-remark-images), [embed videos](/packages/gatsby-remark-embed-video) and plenty of others. Throughout all of these instances, a plugin will examine the Markdown Abstract Syntax Tree (AST) and manipulate content based on certain node types or content in particular nodes.
```

Comment:
```suggestion
In certain instances, a developer may want to customize the content of the Markdown file and extend it functionally in useful ways. For example, use cases such as [adding syntax highlighting](/packages/gatsby-remark-prismjs/), [parsing and creating responsive images](/packages/gatsby-remark-images), [embedding videos](/packages/gatsby-remark-embed-video), and many others. In each of these examples, a plugin will be injected with the Markdown Abstract Syntax Tree (AST) and manipulate content based on certain node types or content in particular nodes.
```

I wonder if we can improve the flow here by the way. It seems that introducing a perhaps confusing/advanced topic such as ASTs would benefit from an example _early on_. I know we do that later--but just thinking aloud.

Not prescriptive here, but curious what you think!

Reply:
my mindset here was to give some examples of plugins and then give a mention for how they are done by introducing an AST. @marcysutton do you have any opinion on this?

Reply:
I think keeping it general like Dustin has written makes the paragraph a little more approachable, with lower cognitive overhead before going into more detail. The only change I would make is to ensure a complete sentence:

```suggestion
In certain instances, a developer may want to customize the content of the Markdown file and extend it functionally in useful ways; for example, use cases such as [adding syntax highlighting](/packages/gatsby-remark-prismjs/), [parsing and creating responsive images](/packages/gatsby-remark-images), [embedding videos](/packages/gatsby-remark-embed-video), and many others. In each of these examples, a plugin will be injected with the Markdown Abstract Syntax Tree (AST) and manipulate content based on certain node types or content in particular nodes.
```

Reply:
Done